src/iocore/cache/AggregateWriteBuffer.cc

```diff
@@ -30,6 +34,23 @@
 
 #include <cstring>
 
+void
+AggregateWriteBuffer::add(Doc const *doc, int approx_size)
+{
+  std::memcpy(this->_buffer + this->_buffer_pos, doc, doc->len);
+  this->_buffer_pos += approx_size;
+  this->add_bytes_pending_aggregation(approx_size);
+}
+
+Doc *
+AggregateWriteBuffer::emplace(int approx_size)
+{
+  Doc *result{new (this->_buffer + this->_buffer_pos) Doc};
+  this->_buffer_pos += approx_size;
+  this->add_bytes_pending_aggregation(-approx_size);
```

Comment:
It should be decreased by `add`. Might have slipped in during a rebase - I'll have to add a test for it, because even the regression tests apparently missed this.

Reply:
See L76-78 - the implementation of `add` is in clear disagreement with the stated requirement.